### DISCUSSION TITLE
Jan: Ollama integration for existing model support

### DISCUSSION THREAD

**Discussion by eckartal** [janhq/jan#3629] [Feature Ideas]
## Overview
- People store models they downloaded from Ollama. In Jan, we should let them run models they already downloaded. This is probably the most commonly asked feature from r/LocalLLaMA.

## Goals
- Implement a feature to detect and import Ollama models into Jan

## A note
- Plus, we can solve this at Cortex-level: https://github.com/orgs/janhq/discussions/3622
- Cortex is a long-term solution, for quick solutions we should consider Ollama support

**Comment by harlley** [comment]
just added a new provider and it worked
<img width="892" height="402" alt="Screenshot 2025-10-25 at 20 05 01" src="https://github.com/user-attachments/assets/9d378306-fbf2-4ec2-8f0e-1cca16eb3c85" />

**Reply by TomLucidor** [reply]
This needs to be an FAQ thing, but does it work with MLX?

**Reply by jade-city2** [reply]
It doesn't support thinking or context length or anything like that.

**Comment by ringe** [comment]
<img width="1279" height="204" alt="bilde" src="https://github.com/user-attachments/assets/046f19bb-6351-4c55-aad3-c5632ae06a28" />

I can't get Ollama to work
```
[GIN] 2026/05/27 - 09:03:50 | 200 |     703.303µs |       127.0.0.1 | GET      "/v1/models"
[GIN] 2026/05/27 - 09:05:31 | 403 |      29.996µs |       127.0.0.1 | POST     "/v1/chat/completions"
[GIN] 2026/05/27 - 09:05:31 | 403 |      11.311µs |       127.0.0.1 | POST     "/v1/chat/completions"
```

Is there some particular setting in Ollama required?

**Reply by ringe** [reply]
CORS problem in the Ollama setup: https://github.com/ollama/ollama/issues/4115#issuecomment-2254177024
